src/Microsoft.VisualStudio.Threading/StackReentrantSemaphoreNestingViolationException.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.Threading
+{
+    using System;
+
+    /// <summary>
+    /// Exception which is thrown by the Reentrant Semaphore when it's reentrancy is violated.
```

Comment:
Suggest rewording this to be something like, "Exception which is thrown when the reentrancy contract of a `<see cref="ReentrantSemaphore"/>` created with `<see cref="ReentrancyMode.Stack"/>` is violated."